diffsync/enum.py

```diff
@@ -47,6 +47,13 @@ class DiffSyncModelFlags(enum.Flag):
     If this flag is set, the model will not be deleted from the target/"to" DiffSync.
     """
 
+    NATURAL_DELETION_ORDER = 0b10000
```

Comment:
to me this looks like the default behavior to be used by default. I can't actually imagine the other behavior use-case. Maybe someone could?

Reply:
It might be a good idea to review known users of this library (https://github.com/topics/diffsync) and confirm their child-deletion behavior. I'm fine with changing the default behavior of DiffSync in a 2.0 version, but if it's relatively low-cost to keep the old behavior as an option, I think it'd be preferable to do so.